File: src/frontend/lib/dto/DrawingAnalysis/DrawingDto.dart
class DrawingDto {
  final String drawingId;
  final String imageURL;
  final String emotionClass;
  final String emotionalInterpretation;
  final String createdAt;
  final List<String> doctorIds;

  DrawingDto({
    required this.drawingId,
    required this.imageURL,
    required this.emotionClass,
    required this.emotionalInterpretation,
    required this.createdAt,
    required this.doctorIds,
  });

  factory DrawingDto.fromJson(Map<String, dynamic> json) {
    return DrawingDto(
      drawingId: (json['drawingId'] ?? '').toString(),
      imageURL: (json['imageURL'] ?? '').toString(),
      emotionClass: (json['emotionClass'] ?? '').toString(),
      emotionalInterpretation: (json['emotionalInterpretation'] ?? '')
          .toString(),
      createdAt: (json['createdAt'] ?? '').toString(),
      doctorIds: (json['doctorIds'] as List? ?? [])
          .map((e) => e.toString())
          .toList(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'drawingId': drawingId,
      'imageURL': imageURL,
      'emotionClass': emotionClass,
      'emotionalInterpretation': emotionalInterpretation,
      'createdAt': createdAt,
      'doctorIds': doctorIds,
    };
  }
}

File: src/backend/src/main/java/com/bouh/backend/model/Dto/DrawingAnalysis/drawingDto.java
package com.bouh.backend.model.Dto.DrawingAnalysis;
import com.google.cloud.firestore.annotation.DocumentId;
import lombok.Data;
import java.time.Instant;
import java.util.List;

@Data //setters,getters and constructors
public class drawingDto {
    @DocumentId
    private String drawingId;

    private String imageURL;
    private String emotionClass;
    private String emotionalInterpretation;
    private Instant createdAt;
    private List<String> doctorIds;
}
